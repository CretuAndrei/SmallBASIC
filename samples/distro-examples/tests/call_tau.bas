' example: using a unit, 0.9
'

Import Tau, PreDef

? "Tau's exported variable: ", tau.expvar
? "Function fooF          : ", tau.fooF("Hi")
? "Procedure fooP         : "
tau.fooP "Hi"

' reverse var-update 
tau.expvar = "message from main"
tau.print_expvar
? tau.expvar
tau.build_ta
? tau.ta
'tau.cerr

rem check system-variables
predef.prsys


