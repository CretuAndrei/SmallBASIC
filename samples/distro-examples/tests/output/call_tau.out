TauChild::initialized

Tau::initilized

Tau's exported variable: 	Tau's exported variable
Function fooF          : 	Tau's fooF(Hi) is here
Procedure fooP         : 
Tau's fooP(Hi) is here
message from main
message from tau
[1,2,3,4]
[1mPredefined Variables[0m
PI     =3.14159265358979
